src/number_theory/divisors.lean

```diff
@@ -398,4 +400,77 @@ begin
   exact nat.lt_succ_of_le (nat.divisor_le (nat.mem_divisors.2 ⟨ha, h⟩))
 end
 
+
+
+/-- # Prime divisors -/
+
+lemma dvd_of_mem_factors {n p : ℕ} (h : p ∈ n.factors) : p ∣ n
+  :=
+begin
+  by_cases hn : n = 0,
+  { convert dvd_zero p },
+  { rwa ←mem_factors_iff_dvd (pos_iff_ne_zero.mpr hn) (prime_of_mem_factors h) }
+end
+
+lemma prime_divisors_eq_to_filter_divisors_prime (n : ℕ) :
+n.factors.to_finset = (divisors n).filter prime
+  :=
+begin
+    by_cases hn : n = 0,
+    { rw [hn, factors_zero, divisors_zero], simp },
+    { ext q,
+      rw [list.mem_to_finset, mem_filter, mem_divisors, mem_factors (pos_iff_ne_zero.mpr hn)],
+      rw [and_comm, and.congr_left_iff],
+      intros hq, simp only [iff_self_and, ne.def], exact imp_intro hn }
+end
+
+/-- The only prime divisor of positive prime power `p^k` is `p` itself -/
+lemma prime_pow_prime_divisor {p k : ℕ} (hk : k ≠ 0) (hp: prime p) :
+  (p^k).factors.to_finset = {p}
+  :=
+begin
+  ext q,
+  rw list.mem_to_finset,
+  rw [mem_factors (pos_iff_ne_zero.mpr (pow_ne_zero k (prime.ne_zero hp))), mem_singleton],
+  split,
+  { rintros ⟨hq1, hq2⟩, exact (prime_dvd_prime_iff_eq hq1 hp).mp (prime.dvd_of_dvd_pow hq1 hq2) },
+  { intros h, rw h, use [hp, dvd_pow_self p hk] },
+end
```

Comment:
This feels like it should much easier if you start with `rw hp.factors_pow`

Reply:
Ah yes, much better.  The main body of the proof becomes `(list.repeat p k).to_finset = {p}`, which ought to exist as a general lemma about `lists` but apparently doesn't.  So I've added this lemma to `data/finset/basic` and then applied it here.